signup-ui/src/pages/shared/mutations.ts

```diff
@@ -55,29 +56,37 @@ export const useVerifyChallenge = () => {
 };
 
 export const useRegister = () => {
+  const { i18n } = useTranslation();
+  const locale = i18n.language == "km" ? "khm" : "eng";
+
   const registerMutation = useMutation<
     RegistrationResponseDto,
     ApiError,
     RegistrationRequestDto
   >({
     mutationKey: keys.registration,
-    mutationFn: (registrationRequestDto: RegistrationRequestDto) =>
-      register(registrationRequestDto),
+    mutationFn: (registrationRequestDto: RegistrationRequestDto) =>{
+      registrationRequestDto.request.locale = locale
+      return register(registrationRequestDto)},
     gcTime: Infinity,
   });
 
   return { registerMutation };
 };
 
 export const useResetPassword = () => {
+  const { i18n } = useTranslation();
+  const locale = i18n.language == "km" ? "khm" : "eng";
```

Comment:
covnerting from 2 letter language code to 3 letter for backend service cannot be handled with a library? We should have solved similar problem in eSignet UI. Can you please connect with @aranaravi for this @mengleang-0090.